web/src/components/shared/UserAds/ChooseRate.tsx

```diff
@@ -0,0 +1,158 @@
+import { useMemo } from 'react';
+import { useDebounce } from 'use-debounce';
+import { Box } from 'web/src/components/ui/Box';
+import { Text } from 'web/src/components/ui/Text';
+import { useFetchRateByParams } from 'web/src/hooks/data/useFetchRate';
+import { TextInput } from 'web/src/components/TextInputCustom/TextInputCustom';
+import { useP2PFiatCurrencies } from 'web/src/hooks/useP2PFiatCurrencies';
+import { useP2PCryptoCurrencies } from 'web/src/hooks/useP2PCryptoCurrencies';
+import { createMoney } from 'web/src/helpers/money';
+import { MoneyFormat } from 'web/src/components/MoneyFormat/MoneyFormat';
+import { Spinner } from 'web/src/components/ui/Spinner';
+import { useAdapterContext } from 'web/src/components/shared/Adapter';
+import * as s from './ChooseRate.css';
+
+export type RateType = 'fixed' | 'floating';
+
+interface Props {
+  type: RateType;
+  fixed: string;
+  percent: string;
+  currency: string | undefined;
+  cryptoCurrency: string;
+  onChangeType: (value: RateType) => void;
+  onChangeFixed: (value: string) => void;
+  onChangePercent: (value: string) => void;
+}
+
+export const ChooseRate = ({
+  type,
+  fixed,
+  percent,
+  currency,
+  cryptoCurrency,
+  onChangeType,
+  onChangeFixed,
+  onChangePercent,
+}: Props) => {
+  const { t } = useAdapterContext();
+  const { getFiatCurrency } = useP2PFiatCurrencies();
+  const { getCryptoCurrency } = useP2PCryptoCurrencies();
+  const ccy = getCryptoCurrency(cryptoCurrency);
+  const [defferedPercent] = useDebounce(percent, 1000);
+  const { data: rateFloating, isValidating: isRateFloatingValidating } = useFetchRateByParams(
+    cryptoCurrency,
+    {
+      currency,
+      percent: defferedPercent ? Number(defferedPercent) : undefined,
+    },
+  );
+  const { data: rateForOne } = useFetchRateByParams(cryptoCurrency, {
+    currency,
+    percent: 0,
+  });
+
+  const rateVariants: Array<{ title: string; value: RateType }> = useMemo(() => {
+    return [
+      { title: t('Fixed rate'), value: 'fixed' },
+      { title: t('Floating rate'), value: 'floating' },
+    ];
+  }, [t]);
+
+  return (
+    <>
+      {rateVariants.map(({ title, value }) => {
+        const isActive = type === value;
+
+        return (
+          <Box
+            key={value}
+            className={isActive ? s.selectedRate : undefined}
+            py="3x"
+            px="4x"
+            borderRadius="1.5x"
+            borderWidth="1x"
+            borderStyle="solid"
+            borderColor={isActive ? 'advertsRateActive' : 'inputBorder'}
+            backgroundColor={isActive ? 'advertsRateActive' : 'transparent'}
+            onClick={() => onChangeType(value)}
+          >
+            <Text fontWeight="strong">{title}</Text>
+
+            <Box display="flex" alignItems="center" mt="3x" gap="3x">
+              {value === 'fixed' ? (
```

Comment:
```suggestion
              {value === 'fixed' && (
```
Замечаю у тебя по коду, то `condition && <Component />`, то `condition ? <Component /> : null`
Старайся делать в одном стиле, хотя бы в пределах одного файла